p2p_wallet/Scenes/Main/Crypto/Components/Crypto Accounts/CryptoAccountsViewModel.swift

```diff
@@ -1,120 +1,60 @@
-        userActionService: UserActionService = Resolver.resolve(),
-        favouriteAccountsStore: FavouriteAccountsDataSource = Resolver.resolve(),
+        interactor: CryptoAccountsInteractorProtocol,
         navigation: PassthroughSubject<CryptoNavigation, Never>
     ) {
-        self.analyticsManager = analyticsManager
-        self.solanaAccountsService = solanaAccountsService
-        self.ethereumAccountsService = ethereumAccountsService
-        self.userActionService = userActionService
-        self.favouriteAccountsStore = favouriteAccountsStore
+        self.interactor = interactor
         self.navigation = navigation
 
         super.init()
-
-        defaultsDisposables.append(Defaults.observe(\.hideZeroBalances) { [weak self] change in
-            self?.hideZeroBalance = change.newValue ?? false
-        })
-
-        bindAccounts()
+        
+        bindToInteractor()
     }
 
     // MARK: - Binding
-
-    private func bindAccounts() {
-        // Ethereum accounts
-        let ethereumAggregator = CryptoEthereumAccountsAggregator()
-        let ethereumAccountsPublisher = Publishers
-            .CombineLatest(
-                ethereumAccountsService.statePublisher,
-                userActionService.actions.map { userActions in
-                    userActions.compactMap { $0 as? WormholeClaimUserAction }
-                }
-            )
-            .map { state, actions in
-                ethereumAggregator.transform(input: (state.value, actions))
-            }
-
-        // Solana accounts
-        let solanaAggregator = CryptoSolanaAccountsAggregator()
-        let solanaAccountsPublisher = Publishers
-            .CombineLatest4(
-                solanaAccountsService.statePublisher,
-                favouriteAccountsStore.$favourites,
-                favouriteAccountsStore.$ignores,
-                $hideZeroBalance
-            )
-            .map { state, favourites, ignores, hideZeroBalance in
-                solanaAggregator.transform(input: (state.value, favourites, ignores, hideZeroBalance))
-            }
-
-        let cryptoAccountsAggregator = CryptoAccountsAggregator()
-        
-        Publishers
-            .CombineLatest(solanaAccountsPublisher, ethereumAccountsPublisher)
-            .map { solanaAccounts, ethereumAccounts in
-                cryptoAccountsAggregator.transform(input: (solanaAccounts, ethereumAccounts))
-            }
-            .receive(on: RunLoop.main)
-            .sink { [weak self] transfer, primary, secondary in
-
-                self?.transferAccounts = transfer
-                self?.accounts = primary
-                self?.hiddenAccounts = secondary
-
-                self?.analyticsManager.log(event: .cryptoClaimTransferredViewed(claimCount: transfer.count))
-            }
+    
+    private func bindToInteractor() {
+        interactor.transferAccountsPublisher
+            .assignWeak(to: \.transferAccounts, on: self)
```

Comment:
For iOS 15 we can use `assign(to: &$transferAccounts)` directly (without .store)